components/list/demo/list-item-sample.js

```diff
@@ -46,11 +48,19 @@ class ListItemSample extends ListItemCheckboxMixin(LitElement) {
 		this._contentId = getUniqueId();
 	}
 
+	connectedCallback() {
```

Comment:
I think this logic needs to be moved into a mixin.

Reply:
hmm, good point.

Reply:
So, I have moved this `connectedCallback` logic hunting for `d2l-list` as a parent to `ListChildMixin`; this seems sufficient. The grid can still be turned on for those implementing their own items by setting `grid-active` on the generic-layout component in whatever way they choose.